### PR TITLE
Update vs2022

### PR DESCRIPTION
**Pull request by wolfv**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by wolfv** [comment]
Apparently I was the first one to try to use VS2022 on conda-forge? :) 

Anyways, I ran into some issues because `vcvars64.bat` is called with the wrong `vcvars_ver`. It needs to be 14.34 now (not 14.33).

You can find the failing CI here: https://github.com/conda-forge/vcpkg-tool-feedstock/pull/41

And Azure logs: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=638918&view=logs&jobId=a70f640f-cc53-5cd3-6cdc-236a1aa90802&j=a70f640f-cc53-5cd3-6cdc-236a1aa90802&t=f5d15007-a01c-5ad8-c9ce-4d519d3b275f

**Comment by h-vetinari** [comment]
> Apparently I was the first one to try to use VS2022 on conda-forge? :)

Maybe ;)

There were also some issues in https://github.com/conda-forge/prismatic_split-feedstock/pull/34 with `cl.exe` not being found. I haven't investigated yet why...

**Comment by wolfv** [comment]
It's the same issue (vcvars.bat doesn't run because the proper version could not be found). This should fix it, too! :)

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ runtime_version:
  # the minor version here can show up in the toolset directory path
  # used by vcvars.bat (see #36), hence we need multiple builds even
  # though they're from the same vsver-line.
- - 14.32.31332
+ - 14.34.31933
```
Comment (isuruf) [inline recipe/conda_build_config.yaml:20]: Can you keep both?

Comment (wolfv) [reply recipe/conda_build_config.yaml]: Done!

**Comment by wolfv** [comment]
I swapped some numbers before, but now they are correct. Would be awesome if we can move forward with this -- then I can try to get `vcpkg-tool` back on track: https://github.com/conda-forge/vcpkg-tool-feedstock/pull/41

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -25,24 +29,28 @@ runtime_version:
...
 # referenceable number.
 update_version:
+ - 4
  - 2
  - 11
  - 8
  - 9
 # This is the version number reported by cl.exe
 cl_version:
+ - 14.34.31933
```
Comment (isuruf) [inline recipe/conda_build_config.yaml:39]:
```suggestion
 - 19.34.31933
```

Comment (wolfv) [reply recipe/conda_build_config.yaml:39]: wow, thanks & sorry for that stupid mistake.

**Comment by wolfv** [comment]
I hope this is good now :)

**Pull request merged**